src/storage/wal_replayer.cpp

```diff
@@ -362,8 +362,9 @@ void WALReplayer::replayUpdateSequenceRecord(const WALRecord& walRecord) const {
     const auto sequenceID = sequenceEntryRecord.sequenceID;
     const auto entry =
         clientContext.getCatalog()->getSequenceCatalogEntry(clientContext.getTx(), sequenceID);
-    entry->replayVal(sequenceEntryRecord.data.usageCount, sequenceEntryRecord.data.currVal,
-        sequenceEntryRecord.data.nextVal);
+    // TODO(Sam): remove unnecessary valuevector allocation
```

Comment:
This looks like an important todo. Let's address it properly.

I see two options:
1. expose an extra interface which correctly replays nextKVal.
2. refactor `nextKVal` to not populate `resultVector`. This doesn't look like an easy way, but it would be a good thing to do.